lib/main.js

```diff
@@ -532,15 +523,15 @@ function GetMML(result) {
 }
 
 //
-//  Creates the speech string and updates the MathML to include it, if needed
+//  Creates dummy speech string and updates the MathML to include it, if needed
 //
 function GetSpeech(result) {
-  if (!data.speakText) return;
-  speech.setupEngine({semantics: true, domain: data.speakRuleset, style: data.speakStyle});
-  result.speakText = speech.toSpeech(result.mml);
+  result.speakText = "Equation";
+  if (data.format !== "MathML") result.speakText = data.math;
   if (!data.mml) return;
   var jax = MathJax.Hub.getAllJax()[0];
-  jax.root.alttext = result.speakText;
+  if (jax.root.alttext) result.speakText = jax.root.alttext;
```

Comment:
What I meant was if alttext is "" rather than non-existent, then it will be overwritten by speakText. Could the case happen that alttext is ever deliberately left empty? If yes, then the test should be refined to check for undefinedness, if no, it's fine.
[This is just a sanity-check. Sometimes careless treatment of semi-boolean values, i.e., "", '', 0, can lead to unintended effects and errors. I ran often enough into that problem, e.g,. when iterating over lists of integers.]